examples/tv-casting-app/linux/CastingShellCommands.cpp

```diff
@@ -80,6 +81,149 @@ void PrintBindings()
+    {
+        ChipLogDetail(Controller, "OnDone CommandSender");
+        if (apCommandSender != nullptr)
+        {
+            chip::Platform::Delete(apCommandSender);
+            ChipLogDetail(Controller, "Cleaned up");
+        }
+    }
+
+} mockCommandSenderDelegate;
+
+CHIP_ERROR SendClusterCommand(chip::NodeId node, chip::EndpointId endpointId, chip::ClusterId clusterId, chip::CommandId commandId,
+                              uint16_t timedRequestTimeoutMs, uint16_t imTimeoutMs, char * json)
+{
+    chip::TLV::TLVWriter writer;
+    chip::TLV::TLVReader reader;
+
+    uint8_t buf[74];
+    uint8_t out_buf[1024];
+
+    Encoding::BufferWriter buf_writer(out_buf, sizeof(out_buf));
+
+    writer.Init(buf);
+    writer.ImplicitProfileId = chip::TLV::kCommonProfileId;
+
+    JsonTLV customJson;
+    ReturnErrorOnFailure(customJson.Parse("SendClusterCommand", json));
+
+    ReturnErrorOnFailure(customJson.Encode(writer, chip::TLV::CommonTag(77)));
+
+    reader.Init(buf, writer.GetLengthWritten());
+    reader.ImplicitProfileId = chip::TLV::kCommonProfileId;
+    reader.Next();
+
+    CastingServer::GetInstance()->SendCommand(mockCommandSenderDelegate, endpointId, clusterId, commandId, reader,
+                                              timedRequestTimeoutMs, imTimeoutMs);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR RunJsonTest(const char * json, const char * testName)
+{
+    chip::TLV::TLVWriter writer;
+    chip::TLV::TLVReader reader;
+
+    uint8_t buf[74];
+    uint8_t out_buf[1024];
+    Encoding::BufferWriter buf_writer(out_buf, sizeof(out_buf));
+
+    writer.Init(buf);
+    writer.ImplicitProfileId = chip::TLV::kCommonProfileId;
+
+    JsonTLV customJson;
+    ReturnLogErrorOnFailure(customJson.Parse(testName, json));
+    ReturnLogErrorOnFailure(customJson.Encode(writer, chip::TLV::CommonTag(77)));
+
+    ChipLogProgress(AppServer, "--------Start Raw Bytes");
+    for (uint32_t i = 0; i < writer.GetLengthWritten(); i++)
+    {
+        if (i != 0 && i % 16 == 0)
+            printf("\n");
+        printf("0x%02X, ", buf[i]);
+    }
+    printf("\n");
+    ChipLogProgress(AppServer, "--------End Raw Bytes");
+
+    reader.Init(buf, writer.GetLengthWritten());
+    reader.ImplicitProfileId = chip::TLV::kCommonProfileId;
+    reader.Next();
+
+    TLV::Debug::Dump(reader, TLV::Json::TLVDumpWriter);
+
+    ChipLogProgress(AppServer, "%s", json);
+    TLV::Json::WriteJSON(reader, buf_writer);
+
+    ChipLogProgress(AppServer, "%s", out_buf);
+
+    Json::Value inputValue;
+    VerifyOrReturnLogError(JsonParser2::ParseCustomArgument(testName, json, inputValue), CHIP_ERROR_INVALID_ARGUMENT);
+
+    Json::Value outputValue;
+    VerifyOrReturnLogError(JsonParser2::ParseCustomArgument(testName, (char *) out_buf, outputValue), CHIP_ERROR_INVALID_ARGUMENT);
+
+    VerifyOrReturnLogError(inputValue == outputValue, CHIP_ERROR_DECODE_FAILED);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR RunJsonTests()
```

Comment:
These tests should be unit tests run during test execution, not a shell command (we can include it in the casting app as well, but a variant of unit tests that are stand alone should exist).